File: 287-find-the-duplicate-number/find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& arr) {
        
        int n=arr.size();
        for(int i=0;i<n;i++){
            if(arr[abs(arr[i])-1]<0){
                return abs(arr[i]);
            }
            else{
               arr[abs(arr[i])-1]=-1*arr[abs(arr[i])-1];
            }
        }

        return 0;
    }
};

File: 287-find-the-duplicate-number/find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        int n = nums.length;

        for(int i=0;i<n;i++){
            if(nums[i]==i+1){
                continue;
            }
            while(nums[i]!=i+1){
                int temp = nums[i];
                int next = nums[nums[i]-1];
                if(nums[i]==next)return nums[i];
                nums[nums[i]-1] = nums[i];
                nums[i] = next;
            }
        }

        return -1;
    }
}